### ISSUE TITLE
pkg-config is unable to find packages even though installed

### ISSUE THREAD

**Issue by Chiiruno** [NixOS/nixpkgs#41825]
## Issue description
`pkg-config --cflags libavcodec libavutil libavformat GraphicsMagick libavcodec libavutil libavformat libswscale`
```
Package libavcodec was not found in the pkg-config search path.
Perhaps you should add the directory containing `libavcodec.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libavcodec' found
Package libavutil was not found in the pkg-config search path.
Perhaps you should add the directory containing `libavutil.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libavutil' found
Package libavformat was not found in the pkg-config search path.
Perhaps you should add the directory containing `libavformat.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libavformat' found
Package GraphicsMagick was not found in the pkg-config search path.
Perhaps you should add the directory containing `GraphicsMagick.pc'
to the PKG_CONFIG_PATH environment variable
No package 'GraphicsMagick' found
Package libavcodec was not found in the pkg-config search path.
Perhaps you should add the directory containing `libavcodec.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libavcodec' found
Package libavutil was not found in the pkg-config search path.
Perhaps you should add the directory containing `libavutil.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libavutil' found
Package libavformat was not found in the pkg-config search path.
Perhaps you should add the directory containing `libavformat.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libavformat' found
Package libswscale was not found in the pkg-config search path.
Perhaps you should add the directory containing `libswscale.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libswscale' found
```

### Steps to reproduce
Attempt the above on the latest nixpkgs with ffmpeg-full and graphicsmagick.

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.48, NixOS, 18.09.git.f13886f (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0.4`
 - channels(root): `"nixos, nixpkgs-18.09pre142380.23f2dfd9f5a"`
warning: Nix search path entry '/home/okina/.nix-defexpr/channels' does not exist, ignoring
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by jtojnar** [comment]
Don't install libraries through nix-env or systemPackages. See https://nixos.wiki/wiki/FAQ/Libraries for details.

**Comment by Chiiruno** [comment]
That was the problem, I had no idea about this. Thanks.
Fixes #39955 properly.
